engine/consensus/matching/sealingtracker/sealing_record.go

```diff
@@ -0,0 +1,97 @@
+package sealingtracker
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// SealingRecord is a record of the sealing status for a specific
+// incorporated Result. It holds information whether the result is sealable,
+// or what is missing to be sealable.
+// Not concurrency safe.
+type SealingRecord struct {
+	ExecutedBlock      *flow.Header             // block the incorporated Result is for
+	IncorporatedResult *flow.IncorporatedResult // the incorporated result
+
+	// SufficientApprovalsForSealing: True iff all chunks in the result have
+	// sufficient approvals
+	SufficientApprovalsForSealing bool
+	// FirstUnmatchedChunkIndex: Index of first chunk that hasn't received
+	// sufficient approval (ordered by chunk index). Optional value: only set
+	// if SufficientApprovalsForSealing == False and nil otherwise.
+	FirstUnmatchedChunkIndex *uint64
```

Comment:
why do we have pointers here?

Reply:
Because they are optional values. `FirstUnmatchedChunkIndex` only has a well-defined value if one of the chunks is missing approvals. If all chunks have sufficient approvals, 

In Java, I would have implemented this using [`Optional`](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html). Did some looking around, and it seems people tend to use pointer types for this in Go (-> [stackoverflow](https://stackoverflow.com/questions/9993178/is-there-a-nice-way-to-simulate-a-maybe-or-option-type-in-go)) ... open to alternative suggestions.

Reply:
To main purpose of the `SealingTracker` is to provide logging information. We don't include the values if they haven't been set by the sealing logic: https://github.com/onflow/flow-go/blob/bedd70477646d65718e1bc391c1e58a3cf75de82/engine/consensus/matching/sealingtracker/sealing_tracker.go#L86-L94